clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -409,6 +409,24 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 instr_free(GLOBAL_DCONTEXT, &instr);
             }
         }
+        if (shard->prev_instr_.instr.addr != 0 /*first*/) {
+            instr_t prev_instr, curr_instr;
+            instr_init(GLOBAL_DCONTEXT, &curr_instr);
+            instr_init(GLOBAL_DCONTEXT, &prev_instr);
+            decode_from_copy(
```

Comment:
Instead of decoding the prev instr again, we can add a bool in per_shard_t for was_prev_instr_syscall and set it at the end of each iteration (and use it in the next iteration).